yeat/config/sample.py

```diff
@@ -85,3 +73,37 @@ def check_reads(self, reads):
                 message = f"Found duplicate read sample '{read}' for '{self.label}'"
                 raise AssemblyConfigError(message)
             self.all_reads.append(read)
+
+    def get_short_readtype(self):
+        illumina = set.intersection(set(self.sample.keys()), set(ILLUMINA_READS))
+        if len(illumina) > 1:
+            message = f"Max of 1 Illumina readtype per sample for '{self.label}'"
+            raise AssemblyConfigError(message)
+        return next(iter(illumina), None)
```

Comment:
Clever use of the default return value for the `next` function here. Would it make the intent clearer to add another case to the conditional?

```python
    def get_short_readtype(self):
        illumina = set.intersection(set(self.sample.keys()), set(ILLUMINA_READS))
        if len(illumina) > 1:
            message = f"Max of 1 Illumina readtype per sample for '{self.label}'"
            raise AssemblyConfigError(message)
        elif len(illumina) == 0:
            return None
        else:
            return next(iter(illumina))
```

Another thing to consider: a better name for the `illumina` variable?